flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/ResourceRequirementsRetrieverTest.java

```diff
@@ -0,0 +1,207 @@
+ */
+
+package org.apache.flink.runtime.scheduler;
+
+import org.apache.flink.api.common.operators.ResourceSpec;
+import org.apache.flink.configuration.MemorySize;
+import org.apache.flink.runtime.clusterframework.types.ResourceProfile;
+import org.apache.flink.runtime.executiongraph.ExecutionGraph;
+import org.apache.flink.runtime.executiongraph.ExecutionJobVertex;
+import org.apache.flink.runtime.executiongraph.TestingExecutionGraphBuilder;
+import org.apache.flink.runtime.instance.SlotSharingGroupId;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionType;
+import org.apache.flink.runtime.jobgraph.DistributionPattern;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobmanager.scheduler.SlotSharingGroup;
+import org.apache.flink.runtime.shuffle.PartitionDescriptor;
+import org.apache.flink.runtime.shuffle.ProducerDescriptor;
+import org.apache.flink.runtime.shuffle.ShuffleDescriptor;
+import org.apache.flink.runtime.shuffle.ShuffleMaster;
+import org.apache.flink.runtime.shuffle.TaskInputsOutputsDescriptor;
+import org.apache.flink.runtime.testtasks.NoOpInvokable;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Unit tests for {@link ResourceRequirementsRetriever}.
+ */
+public class ResourceRequirementsRetrieverTest {
+
+	private static final int OFFSET = 1000;
+
+	private static final ResourceSpec DEFAULT_RESOURCES = ResourceSpec.newBuilder(1.0, 100).build();
+
+	private static final TestShuffleMaster SHUFFLE_MASTER = new TestShuffleMaster();
+
+	private JobGraph jobGraph;
+
+	private ExecutionGraph executionGraph;
+
+	private List<SlotSharingGroup> slotSharingGroups;
+
+	private ResourceRequirementsRetriever resourceRequirementsRetriever;
+
+	@Test
+	public void testJobVertexResourceRequirements() throws Exception {
+		setUp(DEFAULT_RESOURCES);
+
+		final List<JobVertex> jobVertices = jobGraph.getVerticesSortedTopologicallyFromSources();
+		final ExecutionJobVertex source = executionGraph.getJobVertex(jobVertices.get(0).getID());
+		final ExecutionJobVertex map = executionGraph.getJobVertex(jobVertices.get(1).getID());
+		final ExecutionJobVertex sink = executionGraph.getJobVertex(jobVertices.get(2).getID());
+
+		assertEquals(
+			ResourceProfile.fromResourceSpec(DEFAULT_RESOURCES, new MemorySize(0 * OFFSET + 2)),
+			resourceRequirementsRetriever.getJobVertexResourceRequirement(source.getJobVertexId()));
+
+		assertEquals(
+			ResourceProfile.fromResourceSpec(DEFAULT_RESOURCES, new MemorySize(1 * OFFSET + 6)),
+			resourceRequirementsRetriever.getJobVertexResourceRequirement(map.getJobVertexId()));
+
+		assertEquals(
+			ResourceProfile.fromResourceSpec(DEFAULT_RESOURCES, new MemorySize(5 * OFFSET + 0)),
+			resourceRequirementsRetriever.getJobVertexResourceRequirement(sink.getJobVertexId()));
```

Comment:
Will extract the memory computing into a method for tests to directly get expected result, so that we can hide the computing details (like the OFFSET).

Reply:
done.